### PR TITLE
Add  default initialization method for each module

### PR DESCRIPTION
**Pull request by kadisi**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add  default initialization method for each module
```

### CODE REVIEW

**Comment by kadisi** [comment]
> I'd prefer pass the golang context from core.run, then call the cancel in GracefulShutDown to stop all the modules, so we can remove all the CleanUp/Cancel in all modules, thought?

@fisherxu  thanks,  let us move `golang context` to beehiveContext.  and remove the ctx and cancel fields of the module structure。

**Comment by kadisi** [comment]
> I'd prefer pass the golang context from core.run, then call the cancel in GracefulShutDown to stop all the modules, so we can remove all the CleanUp/Cancel in all modules, thought?

@fisherxu  thanks,  let us move `golang context` to beehiveContext.  and remove the ctx and cancel fields of the module structure。

File: edge/test/test.go [hunk 1]
```diff
@@ -229,6 +228,5 @@ func (tm *testManager) Start() {
 	}
 }
 
-func (tm *testManager) Cleanup() {
-	beehiveContext.Cleanup(tm.Name())
+func (tm *testManager) Cancel() {
```
Comment (fisherxu) [inline edge/test/test.go]: Need this func?

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go [hunk 2]
```diff
@@ -16,14 +17,18 @@ const (
 
 var (
 	// singleton
-	context *Context
+	context *BeehiveContext
```
Comment (fisherxu) [inline staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go]: beehiveContext

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package context
 
 import (
+	gcontext "context"
```
Comment (fisherxu) [inline staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go]: Keep `context` here?

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go [hunk 4]
```diff
@@ -16,14 +17,18 @@ const (
 
 var (
 	// singleton
-	context *Context
+	context *BeehiveContext
 	once    sync.Once
 )
 
 // InitContext gets global context instance
 func InitContext(contextType string) {
```
Comment (fisherxu) [inline staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go]: InitBeehiveContext

File: tests/stubs/cloud/controllerstub/module.go [hunk 5]
```diff
@@ -86,8 +86,7 @@ func (cs *ControllerStub) Start(c *context.Context) {
 	downstream.Stop()
 }
 
-// Cleanup resources
-func (cs *ControllerStub) Cleanup() {
+// Cancel resources
+func (cs *ControllerStub) Cancel() {
```
Comment (fisherxu) [inline tests/stubs/cloud/controllerstub/module.go]: Need `cancel` here?

File: tests/stubs/edge/handlerstub/module.go [hunk 6]
```diff
@@ -76,8 +76,7 @@ func (hs *HandlerStub) Start(c *context.Context) {
 	upstream.Stop()
 }
 
-// Cleanup resources
-func (hs *HandlerStub) Cleanup() {
+// Cancel resources
+func (hs *HandlerStub) Cancel() {
```
Comment (fisherxu) [inline tests/stubs/edge/handlerstub/module.go]: ditto

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context.go [hunk 7]
```diff
@@ -1,6 +1,7 @@
 package context
 
 import (
+	gcontext "context"
```
Comment (fisherxu) [inline staging/src/github.com/kubeedge/beehive/pkg/core/context/context.go]: keep `context` here

File: edge/test/test.go (hunk 1 above)
Comment (kadisi) [reply edge/test/test.go]: done

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go (hunk 3 above)
Comment (kadisi) [reply staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go]: package name is `context`, so need add alias

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go (hunk 2 above)
Comment (kadisi) [reply staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go]: ok

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go (hunk 4 above)
Comment (kadisi) [reply staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go]: ok

File: tests/stubs/cloud/controllerstub/module.go (hunk 5 above)
Comment (kadisi) [reply tests/stubs/cloud/controllerstub/module.go]: ok

File: tests/stubs/edge/handlerstub/module.go (hunk 6 above)
Comment (kadisi) [reply tests/stubs/edge/handlerstub/module.go]: done

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context.go (hunk 7 above)
Comment (kadisi) [reply staging/src/github.com/kubeedge/beehive/pkg/core/context/context.go]: change to `gocontext`

Comment (fisherxu) [reply staging/src/github.com/kubeedge/beehive/pkg/core/context/context.go]: ok

**Pull request merged**